test/Constraints/closures.swift

```diff
@@ -66,24 +66,63 @@ func test13811882() {
 // <rdar://problem/21544303> QoI: "Unexpected trailing closure" should have a fixit to insert a 'do' statement
 // <https://bugs.swift.org/browse/SR-3671>
 func r21544303() {
-  let val = true
-  var inSubcall = val
-  {   // expected-error {{expected 'do' keyword to designate a block of statements}} {{3-3=do }}
-  }
+  var inSubcall = true
+  {
+  }  // expected-error {{computed property must have accessors specified}}
   inSubcall = false
 
   // This is a problem, but isn't clear what was intended.
-  var somethingElse = val { // expected-error {{cannot call value of non-function type 'Bool'}}
-  }
+  var somethingElse = true {
+  }  // expected-error {{computed property must have accessors specified}}
   inSubcall = false
 
   var v2 : Bool = false
-  v2 = val
-  {  // expected-error {{expected 'do' keyword to designate a block of statements}}
+  v2 = inSubcall
+  {  // expected-error {{cannot call value of non-function type 'Bool'}} expected-note {{use 'do' keyword to designate a separate block of statements}}
   }
 }
 
 
+// <https://bugs.swift.org/browse/SR-3671>
+func SR3671() {
+  func consume(_ x: Int) {}
+  
+  { consume($0) }(42)
+  ;
+
+  ({ $0(42) } { consume($0) }) // expected-note {{callee is here}}
+  
+  { consume($0) }(42)  // expected-warning {{braces here form a trailing closure separated from its callee by multiple newlines}} expected-error {{cannot call value of non-function type '()'}} expected-note {{use 'do' keyword to designate a separate block of statements}}
+  ;
+  
+  // This is technically a valid call, so nothing goes wrong until (42)
```

Comment:
I wish, but that'd be a source-breaking change. See the bottom of this function.